File: C Programming/problems on recursion/problem12.c
/*
12. Write a recursive program which accept string from user and count number of small characters. 
	
	Input : HElloWOrlD 
	
	Output : 5 
	
	Author : Milind Nalawade
	
	Date : 26/08/2021
	
	Language : C programming
	
*/

#include<stdio.h>
#include<string.h>
int cnt=0,i=0;
int Output(char* arr)
{
	if(*arr=='\0')
	{
		return cnt;
	}
	if(*arr>=97 && *arr<=122)
	{
		cnt++;
	}
	arr++;
	Output(arr);
}	

//Entry point function
int main()
{
	char arr[20];
	int ret=0;
	printf("\n Input : ");
	gets(arr);
	ret=Output(arr);
	printf("\n Output : %d \n",cnt);
	return 0;
}	

File: Python Programming/problems on recursion/problem12.py
'''
12. Write a recursive program which accept string from user and count number of small characters. 
	
	Input : HElloWOrlD 
	
	Output : 5 
	
	Author :Milind Nalawade
	
	Date : 25/08/2021
	
	Language : Python 3.9.0

'''
cnt=0
i=0
#====
#Logic
#====
def Output(str):
	global cnt,i
	if i==len(str):
		return 
	if ord(str[i])>=97 and ord(str[i])<=122:
		cnt+=1
	i+=1	
	Output(str)
		
#=============
#Entry point function
#=============
def main():
	global mul
	str=input("\n Input : ")
	Output(str)
	print("\n Output : {} \n".format(cnt))
	
#=====
#starter
#=====
if __name__=="__main__":
	main()